method.rb

```diff
@@ -0,0 +1,71 @@
+module Enumerable
+    def my_each
+      for i in 0...self.length
+        yield(self[i])
+      end
+    end
+  
+    def my_each_with_index
+      k = 0
+      for i in 0...self.length
+        yield(self[i], k)
+        k += 1
+      end
+    end
+  
+    def my_select(&block)
+      result = []
+      self.each do |element|
+        result << element if block.call(element) == true
+      end
+      result
+    end
+  
+    def my_all?
+      self.my_each { |element| return false if yield(element) == false }
+      true
+    end
+  
+    def my_any?
+      self.my_each { |element| return true if yield(element) }
+      false
+    end
+  
+    def my_none?
```

Comment:
#### `REQUIRED`

:bug: There's a bug to fix. Let's add cases for when a `block` is not given and a `pattern` is passed as an `arg`. Compare with the built-in method.

- When a `block` is not given
```ruby
irb(main):012:0> [1,2,3,4].none?
=> false
irb(main):013:0> [1,2,3,4].my_none?
LocalJumpError (no block given (yield))
```
- When a `pattern` (arg) is passed

```ruby
irb(main):015:0> [1,2,3,4].none?(5)
=> true
irb(main):016:0> [1,2,3,4].my_none?(5)
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html)